### ISSUE TITLE
Transaction/address labeling

### ISSUE THREAD

**Issue by darosior** [wizardsardine/liana#59]
It's a common and useful wallet feature that we'll probably want to have in the future.

**Comment by darosior** [comment]
Kevin and Jimmy agree it's better to label addresses than transactions. Because you want to label payments, not the technical Bitcoin transactions.

It'd be nice to support comments as well. I'll open another issue for that.

**Comment by darosior** [comment]
To iterate on the comment above, i think we should separate the broader "labeling" feature in two sub-features:
- Comments / Descriptions, which is about attaching a few words to a specific transaction or address. For instance "paying back Kevin for the lunch".
- Categories / Tags, or tags, which is about sorting transactions or addresses by common characteristics. For instance "rent", "food", etc..

As mentioned in the comment above i think those two should be tracked separately. I've opened #273 to track the comments / descriptions one.

Pretty much like for comments in #273, i think we need to decide what should be labeled: our addresses? External addresses? Transactions (as in Bitcoin transactions, not payments)?

**Comment by darosior** [comment]
Worth mentioning: https://github.com/bitcoin/bips/blob/master/bip-0329.mediawiki

**Comment by alexwaltz** [comment]
I have a few thoughts on tags vs comments/descriptions
_______

**tags:**
-  plus: bring structure(i mean visually) to the wallet
-  plus: they can also be color coated which helps with easier navigation through TXs but also could make things look aesthetically nicer
- minus: requires making certain assumptions that may not make sense to everyone. Also with predefined categories, almost always there is something that's not quite how I want it to be.

Things to consider re color 
- color-blind people(8% of men, 0.5% of women) - even though I think this could be done in a way that is still visible for color-blind people
- green should be reserved to indicate received bitcoins; red - sent out bitcoins & errors.

An argument to have tags would be, that maybe at the end of the month, you want to see how much you spent on groceries so sorting by this category would make sense, however, I don't think Liana is this type of daily driver wallet, this being more of a requirement for LN wallets and such.
_______
**comments**
- minus: lack of uniformity - requires user to be tidier which is kinda a crazy expectation :)
- plus: fully customizable; should make sense to the user
- minus: long text looks like clutter
_______

Looking at 2 banking apps that I have on my phone, the category(tags) are not displayed in the TX history

As @darosior points out in #273, I also think that it makes more sense for Liana to be coin-centric, so in that case does it even make sense to have tags at all?

TL;DR I think comments/descriptions make more sense for Liana.

**Comment by pythcoiner** [comment]
About comments/labelling,i think it's quite important to have an easy way to remote backup that, but i don't have idea about a solution working for everyone: an easy way for the tech people will be to use something based on git, but will be a pain for 99% non tech users....
any idea of a non-cloud based privacy solution? a nostr DM based to youself?

**Comment by darosior** [comment]
After one year, we've finally merged a first version of labels. We're now experimenting and are tracking tweaks to be made to the labeling system currently in master in https://github.com/wizardsardine/liana/issues/732.
